src/builder/Strings.sol

```diff
@@ -7,7 +7,10 @@ library Strings {
     }
 
     function stringEqIgnoreCase(string memory a, string memory b) internal pure returns (bool) {
-        return keccak256(abi.encodePacked(toLowerCase(a))) == keccak256(abi.encodePacked(toLowerCase(b)));
+        // Need to copy bytes here to prevent unintentional memory changes side effect in strings that are passed in
+        string memory copyA = string(abi.encodePacked(a));
+        string memory copyB = string(abi.encodePacked(b));
```

Comment:
I don't understand why this is needed. Neither this function nor `toLowerCase` modifies `a` or `b`.

Reply:
Actually it does which surprised me too. 

When I called this at the end of the generation (on the resulting action_context), the test failed and I realized all types failed with 'tranfser' != 'TRANFER' and 'BRDIGE_ACTION' != 'bridge_action' and realized it's was here that downcasing all the bytes.

But with thise copy and make string to copy with values,  the test will pass.

Reply:
To be clear, it's actually the `toLowerCase()` that changes the bytes. 
And when I was debating to add this copy value call between toLowerCase() or stringEqIgnoreCase(). 
I decided to put it here, as `stringEgIgnoreCase()` should always not changing the data from input.  
But `toLowerCase()` function, some might think they can just call that to change their string into lowercase.

Reply:
Another way works too as putting copy values into `toLowerCase()`

Reply:
I don't think that's true. `toLowerCase()` shouldn't be modifying the input string, unless I'm misunderstanding something.

Reply:
Hmm...I just tested this locally and it is indeed modifying the input 🤔

Reply:
Yeah I was sort of surprised too. 
But with test failing with `TRANFER != transfer` and adding this the test passed right away sort of pointing to here as the culprit. 

I guess 
```
 bytes memory strBytes = bytes(str);
```
this is probably copy using the same reference as the position where it stores the string.

Reply:
Right, I think it makes sense to do it the way you have then. `toLowerCase` implies modifying the string in-place, so we can leave that as-is